ts/main.ts

```diff
@@ -63,8 +63,145 @@ $entryForm.addEventListener('submit', function (event: Event) {
 
   writeData();
 
+  const renderedEntry = renderEntry(newObj);
+  $entriesList.prepend(renderedEntry);
+  viewSwap('entries');
+  if (
+    data.entries.length > 0 &&
+    !$noEntriesMessage.classList.contains('hidden')
+  ) {
+    $noEntriesMessage.classList.add('hidden');
+  }
+
   $previewImg.setAttribute('src', '../images/placeholder-image-square.jpg');
   // resets the preview image's `src` attribute back to the placeholder image.
   $entryForm.reset();
   // resets form
 });
+
+function renderEntry(entry: Entry): HTMLElement {
+  // <li>
+  //   <div class="row">
+  //     <div class="column-half">
+  //       <img class="list-img" src="${entry.photoUrl}">
+  //     </div>
+  //     <div class="column-half">
+  //       <h3>${entry.title}</h3>
+  //       <p>${entry.notes}</p>
+  //     </div>
+  //   </div>
+  // </li>
+
+  const $entryLi = document.createElement('li');
+
+  const $rowDiv = document.createElement('div');
+  $rowDiv.setAttribute('class', 'row');
+
+  const $colHalfDiv1 = document.createElement('div');
+  $colHalfDiv1.setAttribute('class', 'column-half');
+  const $entryImg = document.createElement('img');
+  $entryImg.setAttribute('class', 'list-img');
+  $entryImg.setAttribute('src', entry.photoURL);
+
+  const $colHalfDiv2 = document.createElement('div');
+  $colHalfDiv2.setAttribute('class', 'column-half');
+  const $entryTitle = document.createElement('h3');
+  $entryTitle.textContent = entry.title;
+  const $entryNotes = document.createElement('p');
+  $entryNotes.textContent = entry.notes;
+
+  $colHalfDiv1.appendChild($entryImg);
+  $colHalfDiv2.appendChild($entryTitle);
+  $colHalfDiv2.appendChild($entryNotes);
+  $rowDiv.appendChild($colHalfDiv1);
+  $rowDiv.appendChild($colHalfDiv2);
+  $entryLi.appendChild($rowDiv);
+
+  return $entryLi;
+}
+
+const $entriesList = document.querySelector(
+  '.entries-list',
+) as HTMLUListElement;
+if (!$entriesList) throw new Error('.entries-list query failed!');
+
+document.addEventListener('DOMContentLoaded', function () {
+  if (data.view === 'entries') {
+    viewSwap('entries');
+  } else if (data.view === 'entry-form') {
+    viewSwap('entry-form');
+  }
+
+  for (let i = 0; i < data.entries.length; i++) {
+    const entry = renderEntry(data.entries[i]);
+    $entriesList.prepend(entry);
+  }
+  if (
+    data.entries.length === 0 &&
+    $noEntriesMessage.classList.contains('hidden')
+  ) {
+    toggleNoEntries();
+  } else if (
+    data.entries.length > 0 &&
+    !$noEntriesMessage.classList.contains('hidden')
+  ) {
+    toggleNoEntries();
+  }
```

Comment:
You should move conditional code _into_ your `toggleNoEntries` function so that you only have to write it once.

Also you should be able to do this with an `if/else`, no need for an `if/else if`